static/js/helpers/windowRegister.js

```diff
@@ -0,0 +1,15 @@
+/* eslint-disable no-console */
+export function register(name, func) {
+  if (typeof window.canonicalWebteam !== "object") {
+    window.canonicalWebteam = {};
+  }
+
+  if (window.canonicalWebteam[name]) {
+    console.error(
+      `Function ${name} has already been declared, please check other modules and make sure that the name is unique.`
+    );
+    return;
+  }
+  window.canonicalWebteam[name] = func;
```

Comment:
I'm OK with exposing modules as globals, so they can be called with parameters, but I think we should do it on build level - so let webpack (or rollup, or else) create a namespace and expose module there, not create our own solution.

In snapcraft.io we use `expose-loader` for that (https://github.com/canonical-websites/snapcraft.io/blob/master/webpack.config.js#L48-L63), but it may be possible directly in webpack without any plugins using `output.library` config (but I haven't used it myself before https://webpack.js.org/configuration/output).

Reply:
Okay, ill extract this into a separate PR. I will work on the cn.ubuntu version as well and we can discuss this in the dev catch up?

I will revert to remove the `window` solution. 

For `output.library` I have found this https://github.com/webpack/webpack/blob/master/examples/multi-part-library/webpack.config.js .

It will need conditional checks on the module name though, otherwise everything defined in the entrypoints will be exposed.

For a multi page setup that seems to be the wrong solution.

Reply:
@b-m-f Not sure what you mean by conditional checks...
Assuming that project has the entrypoint JS file that exports only what should be available publicly it's ok to do it with `output.library` option? I haven't tried that in practice, so I'm not sure how that works.

For the scope of this PR, I'm ok with not exposing anything for now and use it with ids/selectors hardcoded in JS (assuming they are defined as consts somewhere).

Reply:
Oh, but every entrypoint might have different requirements. 

So to only expose certain modules on the window we would need to implement checks :)